src/provider/states.rs

```diff
@@ -0,0 +1,39 @@
+pub(crate) mod download_package;
+pub(crate) mod install_package;
+pub(crate) mod create_config;
+pub(crate) mod waiting_config;
+pub(crate) mod create_service;
+pub(crate) mod download_package_backoff;
+pub(crate) mod setup_failed;
+pub(crate) mod starting;
+pub(crate) mod running;
+pub(crate) mod stopping;
+pub(crate) mod stopped;
+pub(crate) mod failed;
+pub(crate) mod terminated;
+
+
+/// When called in a state's `next` function, exits the current state
+/// and transitions to the Error state.
+#[macro_export]
+macro_rules! transition_to_error {
```

Comment:
Seems unused?

Reply:
I am failry sure that we'll want something like this at some point, but am happy to remove it for now.